azure/activity_logs_monitoring/index.js

```diff
@@ -270,33 +272,46 @@ class EventhubLogHandler {
                         continue;
                     }
                     splitFieldFound = true;
+
                     for (var j = 0; j < recordsToSplit.length; j++) {
                         var splitRecord = recordsToSplit[j];
                         if (typeof splitRecord === 'string') {
                             try {
                                 splitRecord = JSON.parse(splitRecord);
-                            } catch (err) {
-                                splitRecord = { message: splitRecord };
+                            } catch (err) {}
+                        }
+                        var formattedSplitRecord = {};
+                        var temp = formattedSplitRecord;
+                        // re-create the same nested attributes with only the split log
+                        for (var k = 0; k < fields.length; k++) {
+                            if (k === fields.length - 1) {
+                                // if it is the last field, add the split record
+                                temp[fields[k]] = splitRecord;
+                            } else {
+                                temp[fields[k]] = {};
+                                temp = temp[fields[k]];
                             }
                         }
+                        formattedSplitRecord = {
+                            parsed_arrays: formattedSplitRecord
+                        };
 
-                        if (!(splitRecord instanceof Map)) {
-                            // if it's not a map, then just send as a string
-                            splitRecord = {
-                                message: JSON.stringify(splitRecord)
+                        if (config.preserve_fields) {
+                            var newRecord = {};
```

Comment:
Can't remember if the spread operator is available or not. This could technically be `var newRecord = {... originalRecord};`